creation/web_base/utils_xml.sh

```diff
@@ -0,0 +1,317 @@
+          echo "  </result>"
+          echo "  <detail>${my_reason}</detail>"
+      fi
+  else
+      # create a minimal XML file, else
+       echo "  <result>"
+      if [ "${exitcode}" -eq 0 ]; then
+          echo "    <status>OK</status>"
+      else
+          echo "    <status>ERROR</status>"
+          echo "    <metric name=\"failure\" ts=\"$(date --date=@"${glidein_end_time}" +%Y-%m-%dT%H:%M:%S%:z)\" uri=\"local\">Unknown</metric>"
+      fi
+      echo "  </result>
+  <detail>
+    No detail. Could not find source XML file.
+  </detail>"
+  fi
+}
+
+################################
+# Convert base xml to simple xml
+# Arguments:
+#   1: final result
+basexml2simplexml() {
+  local final_result
+  final_result="$1"
+  # augment with node info
+  echo "${final_result}" | awk 'BEGIN{fr=1;}{if (fr==1) print $0}/<operatingenvironment>/{fr=0;}'
+  echo "    <env name=\"client_name\">${client_name}</env>"
+  echo "    <env name=\"client_group\">${client_group}</env>"
+  echo "    <env name=\"user\">$(id -un)</env>"
+  echo "    <env name=\"arch\">$(uname -m)</env>"
+  if [ -e '/etc/redhat-release' ]; then
+      echo "    <env name=\"os\">$(cat /etc/redhat-release)</env>"
+  fi
+  echo "    <env name=\"hostname\">$(uname -n)</env>"
+  echo "${final_result}" | awk 'BEGIN{fr=0;}{if (fr==1) print $0}/<operatingenvironment>/{fr=1;}'
+}
+
+################################
+# Convert simple xml to long xml
+# Arguments:
+#   1: simple final result
+#   2: global result
+simplexml2longxml() {
+  local final_result_simple global_result
+  final_result_simple="$1"
+  global_result="$2"
+  echo "${final_result_simple}" | awk 'BEGIN{fr=1;}{if (fr==1) print $0}/<OSGTestResult /{fr=0;}'
+  if [ "${global_result}" != "" ]; then
+      # subtests first, so it is more readable, when tailing
+      echo '  <subtestlist>'
+      echo '    <OSGTestResults>'
+      echo "${global_result}" | awk '{print "      " $0}'
+      echo '    </OSGTestResults>'
+      echo '  </subtestlist>'
+  fi
+  echo "${final_result_simple}" | awk 'BEGIN{fr=0;}{if (fr==1) print $0}/<OSGTestResult /{fr=1;}/<operatingenvironment>/{fr=0;}'
+  echo "    <env name=\"glidein_factory\">${glidein_factory}</env>"
+  echo "    <env name=\"glidein_name\">${glidein_name}</env>"
+  echo "    <env name=\"glidein_entry\">${glidein_entry}</env>"
+  echo "    <env name=\"condorg_cluster\">${condorg_cluster}</env>"
+  echo "    <env name=\"condorg_subcluster\">${condorg_subcluster}</env>"
+  echo "    <env name=\"glidein_credential_id\">${glidein_cred_id}</env>"
+  echo "    <env name=\"condorg_schedd\">${condorg_schedd}</env>"
+  echo "${final_result_simple}" | awk 'BEGIN{fr=0;}{if (fr==1) print $0}/<operatingenvironment>/{fr=1;}'
+}
+
+################################
+# Add spaces, support to create_xml
+# Globals (r/w):
+#   xml
+# Used:
+#   spaces
+add_spaces(){
+  local c
+  for (( c=1; c<=spaces; c++ ))
+      do
+        xml+=" "
+      done
+}
```

Comment:
consider
```
    xml+=$(printf ' %.0s' $(seq 1 $spaces))
```
consider using it as `xml+=${add_spaces $spaces)` to eliminate global variables. But probably complicates too much the use. Up to you